datalab/middleware.py

```diff
@@ -0,0 +1,16 @@
+from django.core.cache import cache
+
+class CaptureTokenMiddleware:
+    """
+    Middleware to capture the Archive Authorization token from the request headers and store it in the cache
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        token = request.headers.get('Authorization')
+        if token:
+          cache.set('archive_token', token, timeout=None)
```

Comment:
So this cache doesn't differentiate between users. What happens if one user sends a request to the backend, the token gets set, and then another user overwrites the token by sending a request? It seems hard/impossible to guarantee that the token will be set correctly per user with possibly hundreds or thousands of requests coming in at once.

Reply:
I looked into it a bit and it seems like `threading().local` might be an option to store it? I read online some people store the request in the local thread so its accessible throughout the app. Wondering on your thoughts if this could be a solution?

Reply:
Rather than caching it, you can just place it in the request object so the things using it downstream have access to it. You can also just ignore using a middleware and get it out when you need it from the headers in the request.

Reply:
thing is, I use it in a util function called get_fits thats called basically everywhere, so we'd have to pass the request object to basically every function as an argument. So I was looking for a global scope to store it in, unless theres a way to access the request object from anywhere already? Putting it in `local()` seems like a common thing people do

Reply:
I think I would just pass it where it needs to go... I haven't used local stuff before but maybe that is okay too?

Reply:
I'll make a branch of this branch to show what it would look like with the request passed down